packages/core/src/components/Dialog/Dialog.jsx

```diff
@@ -0,0 +1,107 @@
+import AriaModal from 'react-aria-modal';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export const Dialog = function(props) {
+  const {
+    actions,
+    children,
+    escapeExitDisabled,
+    onExit,
+    title,
+    ...modalProps
+  } = props;
+
+  return (
+    <AriaModal
+      dialogClass="ds-c-dialog"
+      escapeExits={!escapeExitDisabled}
+      includeDefaultStyles={false}
+      onExit={onExit}
+      titleId="dialog-title"
+      underlayClass="ds-c-dialog-wrap"
+      {...modalProps}
+    >
+      <header className="ds-c-dialog__header">
+        {title && (
+          <h1 className="ds-h2" id="dialog-title">
+            {title}
+          </h1>
+        )}
+        <button
+          className="ds-c-button ds-c-button--transparent ds-c-dialog__close"
+          onClick={onExit}
+        >
+          Close
+        </button>
+      </header>
+      {children}
```

Comment:
I'd like to consider wrapping the `{children}` declaration in a `<main>` component. This gives another important landmark for screen readers, and follows suit with the `<header>` you're already using. Let's add role="main" as well, for redundant compatibility.